api-service/service/oracle_database_agreements.go

```diff
@@ -658,3 +658,39 @@ func (as *APIService) DeleteHostFromOracleDatabaseAgreement(id primitive.ObjectI
 
 	return as.Database.UpdateOracleDatabaseAgreement(*agreement)
 }
+
+func (as *APIService) DeleteHostFromOracleDatabaseAgreements(hostname string) error {
+	if err := checkHosts(as, []string{hostname}); err != nil {
+		return err
+	}
+
+	listAgreements, err := as.Database.ListOracleDatabaseAgreements()
+	if err != nil {
+		return err
+	}
+
+	for _, la := range listAgreements {
+		agreement, err := as.Database.GetOracleDatabaseAgreement(la.ID)
```

Comment:
Every `la` already contains the list of hosts I think, it could be faster if you use that list and, if you find a match, you load only agreements you are sure you have to update!
Thanks